### DISCUSSION TITLE
Navigation progress bar

### DISCUSSION THREAD

**Discussion by rossanodr** [vercel/next.js#45500] [Help]
Is there any way to make a progress bar when using useRouter?
Something like https://www.npmjs.com/package/nextjs-progressbar
I can't find a way to tell the user that his click on a button actually is taking him somewhere.
Thanks

**Comment by guitheengineer** [comment]
You can play an animation on the `routeChangeStart'` event and end it on the `routeChangeEnd` event
```jsx
  const router = useRouter()
  useEffect(() => {
    router.events.on('routeChangeStart', routeChangeStart);
    router.events.on('routeChangeComplete', routeChangeEnd);
    router.events.on('routeChangeError', routeChangeError);
    return () => {
      router.events.off('routeChangeStart', routeChangeStart);
      router.events.off('routeChangeComplete', routeChangeEnd);
      router.events.off('routeChangeError', routeChangeError);
    };
  }, []);
```

This is how this library is doing actually: https://github.com/apal21/nextjs-progressbar/blob/master/src/index.tsx
